client/components/user-profile.js

```diff
@@ -0,0 +1,307 @@
+import React from "react";
+import { connect } from "react-redux";
+import { useState, useEffect } from "react";
+import { BlockPicker } from "react-color";
+import {
+  Input,
+  Button,
+  Divider,
+  Icon,
+  Card,
+  Image,
+  List,
+  Form,
+  TextArea,
+} from "semantic-ui-react";
+
+export const UserProfile = () => {
+  const [currentBio, SetCurrentBio] = useState("");
+  const [guiltyPleasure, setGuiltyPleasure] = useState("");
+  const [currentMoodImprovement, setCurrentMoodImprovement] = useState({
+    // item: "Music",
+  });
+  const [currentMoodRuin, setCurrentMoodRuin] = useState("");
+
+  const handleKeySubmitBio = (e, input) => {
+    if (e.key === "Enter") {
+      const handleSubmit = (text) => {
+        SetCurrentBio(text);
+      };
+      handleSubmit(input);
+    }
+  };
+  useEffect(() => {}, [handleKeySubmitBio]);
+
+  const handleKeySubmitGP = (e, input) => {
+    if (e.key === "Enter") {
+      const handleSubmit = (text) => {
+        setGuiltyPleasure(text);
+      };
+      handleSubmit(input);
+    }
+  };
+  useEffect(() => {}, [handleKeySubmitGP]);
+
+  const handleKeySubmitImproveMood = (e, input) => {
```

Comment:
This function was going to handle the submission. It does capture the input but I wasn't sure how to add it to setCurrentMoodImprovement as a new value in the initial object.

Reply:
I see, so `currentMoodImprovement` is a string, and we want the result of `setCurrentMoodImprovement` to be either array or object, right? Let's say we go with array, and in the code above we'll do `const [currentMoodImprovement, setCurrentMoodImprovement] = useState([])` to initialize `currentMoodImprovement` as an empty array. Then here, `handleSubmit` can look something like this:
```
const handleSubmit = (text) => {
  const updatedMoodImprovement = [...currentMoodImprovement, text]
  setCurrentMoodImprovement(updatedMoodImprovement)
```
The `...` in `...someIterableObject` is called [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), and using it inside an array will make a copy of that array (this syntax also works on maps)--this way, we're not directly modifying `currentMoodImprovement` (which is `setCurrentMoodImprovement`'s job). We make a copy, along with the new `text` which is going to be an additional mood improvement item, which together create `updatedMoodImprovement` array. Then we can just set that new array as `currentMoodImprovement`. Hope that was helpful.

Really great work by the way!!